src/MySqlDatabase.php

```diff
@@ -0,0 +1,274 @@
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @since [*next-version*]
+     */
+    protected function _getDatabaseName()
+    {
+        return $this->dbName;
+    }
+
+    /**
+     * Sets the database name.
+     *
+     * @since [*next-version*]
+     *
+     * @param string|Stringable $dbName
+     */
+    protected function _setDatabaseName($dbName)
+    {
+        $this->dbName = $this->_normalizeString($dbName);
+    }
+
+    /**
+     * Retrieves the name of the table where migration versions are logged.
+     *
+     * @since [*next-version*]
+     *
+     * @return string|Stringable The name of the version table.
+     */
+    protected function _getLogTableName()
+    {
+        return $this->logTable;
+    }
+
+    /**
+     * Sets the name of the table where migration versions are logged.
+     *
+     * @since [*next-version*]
+     *
+     * @param string|Stringable $logTable The name of the version table.
+     */
+    protected function _setLogTableName($logTable)
+    {
+        $this->logTable = $this->_normalizeString($logTable);
+    }
+
+    /**
+     * Retrieves the name of the version column in the log table.
+     *
+     * @since [*next-version*]
+     *
+     * @return string The name of the version column in the log table.
+     */
+    protected function _getLogTableVersionColumn()
+    {
+        return $this->versionColumn;
+    }
+
+    /**
+     * Sets the name of the version column in the log table.
+     *
+     * @since [*next-version*]
+     *
+     * @param string|Stringable $versionColumn The name of the version column in the log table.
+     */
+    protected function _setLogTableVersionColumn($versionColumn)
+    {
+        $this->versionColumn = $this->_normalizeString($versionColumn);
+    }
+
+    /**
+     * Retrieves the name of the version column in the log table.
+     *
+     * @since [*next-version*]
+     *
+     * @return string The name of the version column in the log table.
+     */
+    protected function _getLogTableStatusColumn()
+    {
+        return $this->statusColumn;
+    }
+
+    /**
+     * Sets the name of the version column in the log table.
+     *
+     * @since [*next-version*]
+     *
+     * @param string|Stringable $statusColumn The name of the version column in the log table.
+     */
+    protected function _setLogTableStatusColumn($statusColumn)
+    {
+        $this->statusColumn = $this->_normalizeString($statusColumn);
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @since [*next-version*]
+     */
+    public function executeSql($sql)
+    {
+        $this->getDbDriver()->execute($sql);
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @since [*next-version*]
+     */
+    public function createDatabase()
+    {
+        $this->executeSql(
+            $this->_formatSql('CREATE SCHEMA IF NOT EXISTS `{db}` DEFAULT CHARACTER SET utf8 ;')
+        );
+        $this->executeSql(
+            $this->_formatSql('USE `{db}`')
```

Comment:
It seems to me like the wrong place for selecting a database.

Reply:
What do you mean?

Reply:
I mean that code that creates a database ("schema" is a synonym" is not probably the code that should choose which DB to use.

Reply:
I understand your concern and why you would w ant this separation. Perhaps you missed the fact that the class `MySqlDatabase`, and also all of its ancestors, represent a **single** DB, not the entire database server.

Reply:
Right, of course. It uses the DB driver object, which represents a specific connection. I withdraw my case,